src/adapters/github.ts

```diff
@@ -309,7 +305,16 @@ class GithubAdapter extends GitAdapter {
   }
 
   protected getRepositoryUrl(repo: IRepo): string {
-    return `git@${gitHubEnterpriseBaseUrl}:${repo.owner}/${repo.name}.git`;
+    const gitHubEnterpriseBaseUrl = process.env.SHEPHERD_GITHUB_ENTERPRISE_BASE_URL ?? 'github.com';
+    const githubProtocol = process.env.SHEPHERD_GITHUB_PROTOCOL ?? 'ssh';
```

Comment:
Curious, what's the reasoning for defaulting to ssh instead of https?

Reply:
The only option right now is ssh. Setting the default to ssh maintains the backwards compatability

Reply:
Hi @jonathanmorley, I've finally got some time to circle back on this.  Changes make sense and you're correct that as the code currently exists, it only supports `ssh`, though it's rather odd as I've certainly used it for `https` without making any changes.  

That said, the changes in this PR explicitly support https protocol and I'll work to land these changes this week.

At the moment I'm working through some dependency upgrades and ESM/CommonJS transpilation issues.  Once complete, this will be a fast follow.